### PR TITLE
Paste path into terminal when drag & dropping file

### LINKED ISSUE #1301: support for dropping a file to insert its path
other terminals support dropping a file from a file manager onto the terminal to insert its path in quotes. this is very convenient when dealing with files that are more easily found through a file manager

it'd be nice if alacritty could support this as well

### PR DESCRIPTION
**Pull request by chrisduerr**
It's now possible to insert the path of a file or directory by dropping
it from the file manager into alacritty.

This fixes #1301.

### CODE REVIEW

**Comment by jwilm** [comment]
Looks like we need a clippy update for the nightly test.

**Comment by jwilm** [comment]
Awesome, nice work. Thanks for implementing this right away!

**Comment by monouser7dig** [comment]
just pulled latest master and tried it on macOS without success, can I give more helpful information?

**Comment by chrisduerr** [comment]
@monouser7dig I think this should work on macos, could you make sure that the event is actually received by putting a `println!("test")` in the relevant part of the code? (https://github.com/jwilm/alacritty/commit/57281363503ddd1ebb373ed5de2b84c79d5aa3aa#diff-3d7686001860aa382f8b9c18554cc138R343).

If it doesn't work on macos, the event should never be received. If that's the case there's probably nothing we can do about that other than reporting it upstream.

**Comment by monouser7dig** [comment]
ok, so it actually works, not sure what I was doing the last time but Alacritty receives the event and inserts the path, nice!

two more things though:
1. the path is only inserted after I klick into the terminal, which I find a bit confusing
2. spaces in the path are not escaped automatically which is a big pain, kitty does not escape them either but Mac terminal does and it makes it much easier/ faster to use

**Comment by chrisduerr** [comment]
@monouser7dig I don't think I've had the issue with the link being inserted only after the terminal has been clicked, however your second point sounds like a good idea.

If you'd like to you could open a new issue to track this.

**Comment by WhyNotHugo** [comment]
I came to the issue tracker looking to open an issue requesting this feature.

It's not working on alacritty 0.6.0 on Linux for me. Has this been reverted, or have I hit a bug?

**Comment by chrisduerr** [comment]
It should work.

**Comment by WhyNotHugo** [comment]
It's not working for me on alacritty 0.6.0, on ArchLinux.

I've run `alacritty -vv`, and it doesn't print anything when I drop the file on it. Shall I open a bug?

**Comment by WhyNotHugo** [comment]
I'm droppping from `nemo`, and drag and drop seems to work between in and  other apps (e.g.: firefox).

**Comment by talbergs** [comment]
Not working with sway. For example `foot` terminal can do this. Could alacritty have this too (on wayland)?

**Comment by WhyNotHugo** [comment]
@talbergs I've reported this not working via #5518, a closed MR doesn't get much visibility and that might help get some help.

File: src/event.rs [hunk 1]
```diff
@@ -338,6 +337,11 @@ impl<N: Notify> Processor<N> {
                         }
 
                         processor.on_focus_change(is_focused);
+                    },
+                    DroppedFile(path) => {
+                        use input::ActionContext;
+                        let path: String = path.to_string_lossy().into();
+                        processor.ctx.write_to_pty(path.into_bytes());
                     }
```
Comment (qstrahl) [inline src/event.rs:345]: Note to future self: Write up a PR allowing the user to configure this behavior through `alacritty.yaml` like any other input.

**Pull request merged**